File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
import math

def solution(progresses, speeds):
    answer = []
    queue = deque()
    for i in range(len(progresses)):
        queue.append(math.ceil((100 - progresses[i]) / speeds[i]))
    
    while queue:
        count = 1 # 첫번째 배포날에 배포하는 개수
        first = queue.popleft() # 맨 처음 원소
        
        while queue and first >= queue[0]:
            count += 1
            queue.popleft()
        
        answer.append(count)
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answerList = new ArrayList<>();
        
        // 각 작업이 완료되기까지 걸리는 일수 계산
        int[] days = new int[progresses.length];
        for (int i = 0; i < progresses.length; i++) {
            days[i] = (int) Math.ceil((100.0 - progresses[i]) / speeds[i]);
        }

        // 배포 단위 계산
        int count = 1;
        int current = days[0];
        
        for (int i=1; i<days.length; i++) {
            if (days[i] <= current) {
                count ++; 
            }
            else {
                answerList.add(count);
                current = days[i];
                count = 1;
            }
        }
        answerList.add(count);

        // List -> int[]
        int[] answer = new int[answerList.size()];
        for (int i = 0; i < answerList.size(); i++) {
            answer[i] = answerList.get(i);
        }

        return answer;
    }
}